File: app/src/main/java/laurent/huberd/radialmenubutton/RadialMenuDemo.java
package laurent.huberd.radialmenubutton;

import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.view.View;

import huberd.laurent.radialbuttonmenu.Animation.BasicAnimationManager;
import huberd.laurent.radialbuttonmenu.RadialMenu;
import huberd.laurent.radialbuttonmenu.Translation.AngleInterval;
import huberd.laurent.radialbuttonmenu.Translation.CircleLayoutManager;

public class RadialMenuDemo extends AppCompatActivity implements RadialMenu.OnRadialMenuStateChangedListener, RadialMenu.OnRadialMenuInteractionListener{

    private View overlayView;
    private RadialMenu radialMenu;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_radial_menu_demo);

        overlayView = findViewById(R.id.overlayView);

        FloatingActionButton floatingMoreActionButton = (FloatingActionButton) findViewById(R.id.floatingMoreActionButton);
        FloatingActionButton floatingCameraActionButton = (FloatingActionButton) findViewById(R.id.floatingCameraActionButton);
        FloatingActionButton floatingGalleryActionButton = (FloatingActionButton) findViewById(R.id.floatingGalleryActionButton);
        FloatingActionButton floatingTextActionButton = (FloatingActionButton) findViewById(R.id.floatingTextActionButton);

        radialMenu = new RadialMenu(floatingMoreActionButton, new CircleLayoutManager(new AngleInterval(30, 150)), new BasicAnimationManager());
        radialMenu.setOnRadialMenuInteractionListener(this);
        radialMenu.setOnRadialMenuStateChangeListener(this);

        radialMenu.addSecondaryButton(floatingTextActionButton);
        radialMenu.addSecondaryButton(floatingCameraActionButton);
        radialMenu.addSecondaryButton(floatingGalleryActionButton);
    }

    @Override
    public void onOptionSelected(int i) {
        System.out.println(i);
    }

    @Override
    public void onOptionHover(int i) {
        System.out.println(i);
    }

    @Override
    public void onStateChanged(RadialMenu.RadialMenuState oldState, RadialMenu.RadialMenuState newState) {
        switch (newState) {
            case EXTENDED:
                break;
            case MINIMIZED:
                break;
            case ANIMATING:
                toggleOverlay(oldState);
                break;

        }
    }

    private void toggleOverlay(RadialMenu.RadialMenuState oldState) {
        switch (oldState) {
            case EXTENDED:
                overlayView.animate().alpha(0).setDuration(150).withEndAction(new Runnable() {
                    @Override
                    public void run() {
                        overlayView.setOnClickListener(null);
                        overlayView.setVisibility(View.GONE);
                    }
                });
                break;
            case MINIMIZED:
                overlayView.clearAnimation();
                overlayView.setVisibility(View.VISIBLE);
                overlayView.setAlpha(0);
                overlayView.animate().alpha(1).setDuration(150).start();

                overlayView.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        if (radialMenu.getRadialMenuState() == RadialMenu.RadialMenuState.EXTENDED) {
                            radialMenu.hide();
                        }
                    }
                });
                break;
        }
    }
}

File: app/src/main/java/laurent/huberd/radialmenubutton/RadialMenuDemo.kt
package laurent.huberd.radialmenubutton

import android.os.Bundle
import android.support.design.widget.FloatingActionButton
import android.support.v7.app.AppCompatActivity
import android.view.View

import huberd.laurent.radialbuttonmenu.Animation.BasicAnimationManager
import huberd.laurent.radialbuttonmenu.RadialMenu
import huberd.laurent.radialbuttonmenu.Translation.AngleInterval
import huberd.laurent.radialbuttonmenu.Translation.CircleLayoutManager

class RadialMenuDemo : AppCompatActivity(), RadialMenu.OnRadialMenuStateChangedListener, RadialMenu.OnRadialMenuInteractionListener {

    private var overlayView: View? = null
    private lateinit var radialMenu: RadialMenu

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_radial_menu_demo)

        overlayView = findViewById(R.id.overlayView)

        val floatingMoreActionButton = findViewById(R.id.floatingMoreActionButton) as FloatingActionButton
        val floatingCameraActionButton = findViewById(R.id.floatingCameraActionButton) as FloatingActionButton
        val floatingGalleryActionButton = findViewById(R.id.floatingGalleryActionButton) as FloatingActionButton
        val floatingTextActionButton = findViewById(R.id.floatingTextActionButton) as FloatingActionButton

        radialMenu = RadialMenu(mainButton = floatingMoreActionButton,
                layoutManager = CircleLayoutManager(AngleInterval(30f, 150f)),
                animationManager = BasicAnimationManager())
        radialMenu.onRadialMenuInteractionListener = this
        radialMenu.onRadialMenuStateChangeListener = this

        radialMenu.addSecondaryButton(floatingTextActionButton)
        radialMenu.addSecondaryButton(floatingCameraActionButton)
        radialMenu.addSecondaryButton(floatingGalleryActionButton)
    }

    override fun onOptionSelected(i: Int) {
        println(i)
    }

    override fun onOptionHover(i: Int) {
        println(i)
    }

    override fun onStateChanged(oldState: RadialMenu.RadialMenuState, newState: RadialMenu.RadialMenuState) {
        when (newState) {
            RadialMenu.RadialMenuState.EXTENDED -> {
            }
            RadialMenu.RadialMenuState.MINIMIZED -> {
            }
            RadialMenu.RadialMenuState.ANIMATING -> toggleOverlay(oldState)
        }
    }

    private fun toggleOverlay(oldState: RadialMenu.RadialMenuState) {
        when (oldState) {
            RadialMenu.RadialMenuState.EXTENDED -> overlayView!!.animate().alpha(0f).setDuration(150).withEndAction {
                overlayView!!.setOnClickListener(null)
                overlayView!!.visibility = View.GONE
            }
            RadialMenu.RadialMenuState.MINIMIZED -> {
                overlayView!!.clearAnimation()
                overlayView!!.visibility = View.VISIBLE
                overlayView!!.alpha = 0f
                overlayView!!.animate().alpha(1f).setDuration(150).start()

                overlayView!!.setOnClickListener {
                    if (radialMenu.radialMenuState == RadialMenu.RadialMenuState.EXTENDED) {
                        radialMenu.hide()
                    }
                }
            }
            RadialMenu.RadialMenuState.ANIMATING -> {
                
            }
        }
    }
}
